src/NextFederationPlugin.js

```diff
@@ -317,81 +256,101 @@ class ChildFederation {
+      //           resolve(childCompilation.assets);
+      //         }
+      //       );
+      //     });
+      //
+      //   } else {
+      //
+      //       TODO: improve this
+      //       childAssets = new Promise((resolve, reject) => {
+      //         fs.readdir(
+      //           path.join(childCompiler.context, '.next/ssr'),
+      //           function (err, files) {
+      //             if (err) {
+      //               reject('Unable to scan directory: ' + err);
+      //               return;
+      //             }
+      //
+      //             const allFiles = files.map(function (file) {
+      //               return new Promise((res, rej) => {
+      //                 fs.readFile(
+      //                   path.join(childCompiler.context, '.next/ssr', file),
+      //                   (err, data) => {
+      //                     if (err) rej(err);
+      //                     compilation.assets[path.join('static/ssr', file)] = new compiler.webpack.sources.RawSource(data)
+      //                     res();
+      //                   }
+      //                 );
+      //               });
+      //             });
+      //             Promise.all(allFiles).then(resolve).catch(reject)
+      //           }
+      //         );
+      //       });
+      //   }
+      // }
       // on main compiler add extra assets from server output to browser build
-      compilation.hooks.additionalAssets.tapPromise(CHILD_PLUGIN_NAME, () => {
-        console.log(compiler.options.name);
-        console.log('in additional assets hook for main build', childAssets);
+      // compilation.hooks.additionalAssets.tapPromise(CHILD_PLUGIN_NAME, () => {
+      //   console.log('additional hooks', compiler.options.name);
+      //   console.log('in additional assets hook for main build');
+      //   return childAssets
+      // });
 
-        return childAssets
-      });
+      // cache the serer compiler instance, we will run the server child compiler during the client main compilation
+      // we need to do this because i need access to data from the client build to inject into the server build
+      // in prod builds, server build runs first, followed by client build
+      // in dev, client build runs first, followed by server build
+      childCompilers[compiler.options.name] = childCompiler;
 
-      if (compiler.options.mode === 'development') {
+
+      if(isDev) {
+        // in dev, run the compilers in the order they are created (client, server)
         childCompiler.run((err, stats) => {
-          // moveFilesToClientDirectory(isServer, stats);
           if (err) {
             console.error(err);
             throw err
           }
         });
-      } else {
+        // in prod, if client
+      } else if(!isServer) {
+        //wrong hook for this
+        // add hook for additional assets to prevent compile from sealing.
+        compilation.hooks.additionalAssets.tapPromise(CHILD_PLUGIN_NAME, () => {
```

Comment:
@alexander-akait this is another area i could use some advise on. 

I need client build to exist, so i can access stats/manifest. 
So i "cache" server child compiler instance and execute it under the client main compilation